### PR TITLE
Release/0.8.2

### PR DESCRIPTION
**Pull request by SergioEstevao**
Update pod spec and some build configuration.

### CODE REVIEW

File: wpxmlrpc.podspec [hunk 1]
```diff
@@ -1,16 +1,15 @@
 Pod::Spec.new do |s|
   s.name         = "wpxmlrpc"
-  s.version      = "0.8"
+  s.version      = "0.8.2"
   s.summary      = "Lightweight XML-RPC library."
   s.homepage     = "https://github.com/wordpress-mobile/wpxmlrpc"
-  s.license      = { :type => 'MIT', :file => 'LICENSE.md' }
-  s.author       = "WordPress"
+  s.license      = { :type => 'GPLv2', :file => 'LICENSE.md' }
+  s.author       = {"WordPress", "Automattic" => "mobile@automattic.com"}
```
Comment (koke) [inline wpxmlrpc.podspec]: Why did you change the license?

File: wpxmlrpc.podspec [hunk 2]
```diff
@@ -1,16 +1,15 @@
...
   s.source       = { :git => "https://github.com/wordpress-mobile/wpxmlrpc.git", :tag => "#{s.version}" }
   s.source_files = 'WPXMLRPC'
-  s.public_header_files = [ 'WPXMLRPC/WPXMLRPC.h', 'WPXMLRPC/WPXMLRPCEncoder.h', 'WPXMLRPC/WPXMLRPCDecoder.h' ]
   s.libraries    = 'iconv'
   s.requires_arc = true
 
-  s.ios.deployment_target = '5.0'
-  s.osx.deployment_target = '10.7'
+  s.ios.deployment_target = '7.0'
+  s.osx.deployment_target = '10.9'
```
Comment (koke) [inline wpxmlrpc.podspec]: Since we didn't use the new Base64 methods, do we still need to bump the deployment target?

File: wpxmlrpc.podspec [hunk 3]
```diff
@@ -1,16 +1,15 @@
...
+  s.version      = "0.8.2"
   s.summary      = "Lightweight XML-RPC library."
   s.homepage     = "https://github.com/wordpress-mobile/wpxmlrpc"
-  s.license      = { :type => 'MIT', :file => 'LICENSE.md' }
-  s.author       = "WordPress"
+  s.license      = { :type => 'GPLv2', :file => 'LICENSE.md' }
+  s.author       = {"WordPress", "Automattic" => "mobile@automattic.com"}
   s.source       = { :git => "https://github.com/wordpress-mobile/wpxmlrpc.git", :tag => "#{s.version}" }
   s.source_files = 'WPXMLRPC'
-  s.public_header_files = [ 'WPXMLRPC/WPXMLRPC.h', 'WPXMLRPC/WPXMLRPCEncoder.h', 'WPXMLRPC/WPXMLRPCDecoder.h' ]
```
Comment (koke) [inline wpxmlrpc.podspec]: Why was this removed?

File: wpxmlrpc.podspec (hunk 1 above)
Comment (SergioEstevao) [inline wpxmlrpc.podspec]: I was under the assumption that all wordpress related projects needed to be under GPLv2. But if I'm wrong I can revert this.

File: wpxmlrpc.podspec (hunk 3 above)
Comment (SergioEstevao) [inline wpxmlrpc.podspec]: Because I changed the structure of the test project to use development pod instead of directly including the files, I need access to all h file in order to make the tests compile.

File: wpxmlrpc.podspec (hunk 2 above)
Comment (SergioEstevao) [inline wpxmlrpc.podspec]: You are right, let me revert this.

File: wpxmlrpc.podspec (hunk 1 above)
Comment (koke) [inline wpxmlrpc.podspec]: I think generally that's correct, but in this case this started as a fork of an existing library. While it's possible to change the license from MIT to GPL, a) I don't think we should in this case, b) it should be discussed first

File: wpxmlrpc.podspec (hunk 3 above)
Comment (koke) [inline wpxmlrpc.podspec]: This exposes a few header files that aren't documented, and even if they where they just make the API more confusing since they're not relevant to users of this library.

This brings up an interesting discussion about testing, and I was talking with @astralbodies about this yesterday. 

**Should we only test our public interfaces?**

If this is the case, we'd need to refactor our tests so they only use the public headers. So, instead of calling `WPXMLRPCDataCleaner` directly, we'd test `WPXMLRPCDecoder` to see if it can handle _dirty_ data.

If we consider that it's simpler/better to test these internal classes separately, then the test project should test the sources directly and not the pod.

In no case we should expose internal classes to the public _just_ to make testing more convenient.

**Comment by koke** [comment]
:shipit:
